clients/wallet/src/stellar_wallet.rs

```diff
@@ -47,9 +48,21 @@ impl StellarWallet {
 
 		let submission_result = horizon_client
 			.submit_transaction(envelope.clone(), self.is_public_network)
-			.await;
+			.await
+			.map(|resp| {
+				// use the paging token to save the latest cursor
+				if let Err(e) = self.cache.save_cursor(resp.paging_token) {
```

Comment:
I am not sure if it makes sense to save the cursor here. I think this might introduce unwanted side effects. Especially considering my main comment about the scenario with the race condition.

I would only write a new paging token to the cache after fetching requests.

Reply:
removed